app/controllers/core_data_connector/sso_controller.rb

```diff
@@ -0,0 +1,88 @@
+require 'typhoeus'
+require 'jwt'
+require 'jwt_auth/configuration'
+require 'jwt_auth/json_web_token'
+require 'uri'
+
+module CoreDataConnector
+  class SsoController < ApplicationController
+    DEFAULT_TOKEN_EXPIRATION = 24
+
+    # No user authentication
+    skip_before_action :authenticate_request
+
+    def login
+      base_url = ENV['SSO_TOKEN_URL']
+
+      form_data = {
+        code: request.params['code'],
+        client_id: ENV['SSO_CLIENT_ID'],
+        client_secret: ENV['SSO_CLIENT_SECRET'],
+        grant_type: 'authorization_code',
+        redirect_uri: request.original_url.split('?')[0]
+      }
+
+      token_req = Typhoeus::Request.new(
+        base_url,
+        headers: {'Content-Type'=> "application/x-www-form-urlencoded"},
+        method: :post,
+        body: form_data
+      )
+
+      token_res = token_req.run
+      parsed = JSON.parse(token_res.body)
+
+      decoded = JWT.decode(parsed['access_token'], nil, false)[0]
+
+      sso_metadata = HashWithIndifferentAccess.new decoded
+
+      sso_user = find_sso_user(sso_metadata)
+
+      unless sso_user
+        render json: { error: 'User not found.' }, status: :bad_request
+      end
+      
+      user_serializer = JwtAuth.config.user_serializer.constantize
```

Comment:
Any reason we're using the `JwtAuth` configured serializer here and not `UsersSerializer` directly? I know we probably want the user record to be serialized the same way in both situations, but it seems a little weird to differ to a gem for that.

Reply:
I got the impression that `JwtAuth` makes the model it's connected to configurable and wanted to keep the SSO handling consistent with how `JwtAuth` is configured. If that's not important I can point directly to the user serializer.

Reply:
It's true that `JwtAuth` is configureable, model and serializer. But it's Core Data that does the configuration [here](https://github.com/performant-software/core-data-connector/blob/e680acee1b00e36a5b1bc14d5f51733e552114c5/lib/core_data_connector/engine.rb#L15). It's just a little weird that Core Data is then going back to `JwtAuth` to get the serializer, when Core Data determined the serializer in the first place.